clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1754,8 +1824,10 @@ bool CompilerInvocation::ParseCodeGenArgs(CodeGenOptions &Opts, ArgList &Args,
             : codegenoptions::DebugTemplateNamesKind::Mangled);
   }
 
-  if (!Opts.ProfileInstrumentUsePath.empty())
-    setPGOUseInstrumentor(Opts, Opts.ProfileInstrumentUsePath, Diags);
+  if (!Opts.ProfileInstrumentUsePath.empty()) {
+    auto FS = createBaseFS(FSOpts, CASOpts, Diags, nullptr);
+    setPGOUseInstrumentor(Opts, Opts.ProfileInstrumentUsePath, *FS, Diags);
```

Comment:
This doesn't seem it would work with include tree, `createBaseFS` would create a `llvm::vfs::getRealFileSystem()` in that case so this would try to access the file from the real filesystem. A couple of suggestions:
* We could move the `cas::createIncludeTreeFileSystem(*Root)` call to be in `createBaseFS`.
* In the test cases make sure to delete the files from the real filesystem, before doing the caching invocation that uses CAS as input, to ensure the files are not read directly from the disk.

Reply:
Sounds good. How does path prefix remapping for include tree works? I assume it is still needed for the file in the IncludeTreeFileSystem but it seems it just crashes.

Reply:
> How does path prefix remapping for include tree works? I assume it is still needed for the file in the IncludeTreeFileSystem but it seems it just crashes.

The mechanism is not in OSS yet, don't try to test it from here.